framer/GestureInputRecognizer.coffee

```diff
@@ -0,0 +1,448 @@
+Utils = require "./Utils"
+
+GestureInputLongPressTime = 0.5
+GestureInputDoubleTapTime = 0.25
+GestureInputSwipeThreshold = 30
+GestureInputEdgeSwipeDistance = 30
+GestureInputVelocityTime = 0.1
+GestureInputForceTapDesktop = MouseEvent.WEBKIT_FORCE_AT_FORCE_MOUSE_DOWN
+GestureInputForceTapMobile = 0.7
+GestureInputForceTapMobilePollTime = 1/30
+
+{DOMEventManager} = require "./DOMEventManager"
+
+class exports.GestureInputRecognizer
+	
+	constructor: ->
+		@em = new DOMEventManager()
+		@em.wrap(window).addEventListener("touchstart", @touchstart)
```

Comment:
When I saw this, I thought it was a bit extreme. But after giving some thought, I don't see any other way to do it. I checked Hammer code and they use this technique too (although they add some extra elements besides `window`)